src/main/kotlin/org/jetbrains/research/libsl/visitors/ExpressionVisitor.kt

```diff
@@ -211,83 +224,86 @@ class ExpressionVisitor(
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "u" -> UnsignedInt32Literal(
-                    ctx.text.dropLast(1).toUInt(),
-                    "u",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "L" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toLong(),
-                    "L",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "uL" -> UnsignedInt64Literal(
-                    ctx.text.dropLast(2).toULong(),
-                    "uL",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                else -> throw IllegalArgumentException("Incorrect integer suffix")
-            }
+                ctx.text.dropLast(1).toByte(),
+                "b",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("us")) {
+            return UnsignedInt16Literal(
+                ctx.text.dropLast(2).toUShort(),
+                "us",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("s")) {
+            return IntegerLiteral(
+                ctx.text.dropLast(1).toShort(),
+                "s",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("u")) {
+            return UnsignedInt32Literal(
+                ctx.text.dropLast(1).toUInt(),
+                "u",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("ul")) {
+            return UnsignedInt64Literal(
+                ctx.text.dropLast(2).toULong(),
+                "uL",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+        } else if (num.endsWith("l")) {
+            num = num.dropLast(1)
+            return IntegerLiteral(
+                convertBinHexOctToPrimitives(num, "l"),
+                "L",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
         }
+        return IntegerLiteral(
+            convertBinHexOctToPrimitives(num, "i"),
+            null,
+            posGetter.getCtxPosition(context.fileName, ctx)
+        )
     }
 
+    fun convertBinHexOctToPrimitives(num: String, type: String): Number {
+        val isInt = "i".equals(type)
+        val result: Number = if (num.startsWith(HEX_PREFIX)) {
+            if (isInt) parseInt(num.drop(2), 16) else parseLong(num.drop(2), 16)
+        } else if (num.startsWith(BIN_PREFIX)) {
+            if (isInt) parseInt(num.drop(2), 2) else parseLong(num.drop(2), 2)
+        } else if (num.startsWith(OCT_PREFIX) && num.length > 1) {
+            if (isInt) parseInt(num.drop(1), 8) else parseLong(num.drop(1), 8)
+        } else {
+            if (isInt) num.toInt() else num.toLong()
+        }
```

Comment:
May be something like this will be a bit better?
```kotlin
return = when {
    num.startsWith(HEX_PREFIX) -> isInt ? parseInt(num.drop(2), 16) : parseLong(num.drop(2), 16)
    num.startsWith(OCT_PREFIX) -> isInt ? parseInt(num.drop(2), 8) : parseLong(num.drop(2), 8)
    num.startsWith(BIN_PREFIX) -> isInt ? parseInt(num.drop(2), 2) : parseLong(num.drop(2), 2)
    _ -> isInt ? num.toInt() : num.toLong()
}
```

Reply:
Fixed